packages/test/test-version-utils/src/test/minCompatVersion.spec.ts

```diff
@@ -0,0 +1,87 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { execSync } from "child_process";
+import { CompatKind } from "../../compatOptions.cjs";
+import { isCompatVersionBelowMinVersion } from "../compatConfig.js";
+
+describe("Minimum Compat Version", () => {
+	const allVersionsFromNpm = execSync(`npm show fluid-framework versions --json`, {
+		encoding: "utf-8",
+	});
+	const allVersions: string[] = JSON.parse(allVersionsFromNpm);
+	const latestVersion = allVersions[allVersions.length - 1];
+
+	it("bad min compat string", () => {
+		const invalidString = "invalid string";
+		try {
+			isCompatVersionBelowMinVersion(invalidString, {
+				name: `test`,
+				kind: CompatKind.None,
+				compatVersion: "2.0.0-internal.8.0.0",
+			});
+			assert.fail("test should fail");
+		} catch (error: any) {
+			assert.strictEqual(
+				error.message,
+				`Error while running: npm v @fluidframework/container-loader@"${invalidString}" version --json`,
+			);
+		}
+	});
+
+	for (let i = 1; i < 9; i++) {
+		it(`compatVersion N-${i} < latest version ${latestVersion}`, () => {
+			assert.strictEqual(
+				isCompatVersionBelowMinVersion(latestVersion, {
+					name: `test`,
+					kind: CompatKind.None,
+					compatVersion: -i,
+				}),
+				true,
+				`N-${i} is not lower than min version`,
+			);
+		});
+	}
+
+	it("cross compat", () => {
+		assert.strictEqual(
```

Comment:
nit: These 4 different asserts may be better as separate test cases as they are testing different aspects of the API. At least you should add a comment / fail message to each that will help with debugging / understanding.